test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,110 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	nmstate     nmstatev1beta1.NMState
+	nmstateKey  = types.NamespacedName{Namespace: "nmstate", Name: "nmstate"}
+	handlerKey  = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-handler"}
+	operatorKey = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-operator"}
+	webhookKey  = types.NamespacedName{Namespace: "nmstate", Name: "nmstate-webhook"}
+)
+
+func installNMState() {
+	err := framework.Global.Client.Get(context.TODO(), nmstateKey, &nmstate)
```

Comment:
Passing the content of nmstate on the side as a global variable is a terrible practice. It is hard to connect all dots and spot potential issue where the variable does not contain what we expect.

Would you please pass both the key and the content as a parameter to this function? Unless you can make those vars into constants, but I'm afraid that's not possible with structs.